crates/sui-e2e-tests/tests/zklogin_tests.rs

```diff
@@ -193,11 +193,12 @@ async fn test_auth_state_creation() {
         .build()
         .await;
 
+    // Now wait until the auth state object is created, ie. AuthenticatorStateUpdate transaction happened.
+    test_cluster.wait_for_authenticator_state_update().await;
```

Comment:
i'm pretty sure this makes the wait_for_protocol_version useless, since we can't still be in v23 when this completes. in which case i don't understand what this fixes

Reply:
my understanding is the authenticator state update happens before the protocol reaches 24. the purpose of this test is to make sure we are able to reach at a point where authenticator state exists. and since zklogin test runs very long, i have isolated the transactional testing into `async fn zklogin_end_to_end_test() {`